### DISCUSSION TITLE
docker compose file

### DISCUSSION THREAD

**Discussion by comfriends** [mindsdb/mindsdb#1272] [Get help]
I want to run mindsdb through the docker compose file instead of the <code>docker run</code> command.

![캡처2](https://user-images.githubusercontent.com/62046692/119284245-2f44c480-bc7a-11eb-8f7d-6c3ba4525300.PNG)

Are there any parts to be modified?

**Comment by ZoranPandovski** [comment]
It looks ok. The only thing you can remove is `-- config=/root/mindsdb_config.json` since mindsdb will use the default parameters and later on if you want to modify them, the best way will be through the GUI. Is there a particular why you want to use compose, or you will start the database as a service in compose too?

**Reply by comfriends** [reply]
I use clickhouse and run it with docker.
I want to run mindsdb on the same stack.
And if the container dies, it automatically reruns.
 Can set up the network.

**Reply by ZoranPandovski** [reply]
Great. If you have more questions or experience issues let us know.

**Reply by ZoranPandovski** [reply]
Hello @combff, did you manage to use MindsDB and Clickhouse?

**Reply by comfriends** [reply]
Sorry for my belated reply. gui works well.
But the query throws an error like this:

![캡처](https://user-images.githubusercontent.com/62046692/120959815-8f1c8e80-c795-11eb-9957-c99cf8107b7a.PNG)

Is only gui action possible?

**Reply by ZoranPandovski** [reply]
No, it should work for both clickhouse-client and GUI too. The `test1` model was trained from the GUI right?

**Reply by comfriends** [reply]
yes i trained in gui.

**Reply by comfriends** [reply]
How can I get rid of the error?

**Reply by ZoranPandovski** [reply]
Can you try to run the  **SELECT * FROM mindsdb.predictors;** ? Or again it is the same error? Also, can you share what version of MindsDB you are using, because the `0.0.0.0` MySQL connection is from old versions of Mindsdb.

**Reply by comfriends** [reply]
same error comes up.

**Reply by comfriends** [reply]
I downloaded the latest version from Docker Hub.

**Reply by ZoranPandovski** [reply]
I guess you are extending the default config then. Can you share the config.json configuration?

**Reply by comfriends** [reply]
![mindsdb_config](https://user-images.githubusercontent.com/62046692/121616641-39e6c280-ca9e-11eb-931f-e5a8541f9f79.PNG)

**Reply by ZoranPandovski** [reply]
Can you try to add the mysql `host` to be 127.0.0.1 and random `password`. Apart from that all looks good.

**Reply by comfriends** [reply]
Is it correct to do this?

![mindsdb_config1](https://user-images.githubusercontent.com/62046692/121863192-8e0bd400-cd36-11eb-9155-cad2f358b499.PNG)

**Reply by comfriends** [reply]
But I get an error.
![에러](https://user-images.githubusercontent.com/62046692/121862954-4c7b2900-cd36-11eb-8f80-638eec7875f7.PNG)

**Reply by ZoranPandovski** [reply]
You are hosting MindsDB locally, but the ClickHouse server is in remote machine? If yes, you need to make a port forwarding so you can connect your localhost to the remote server e.g ` ssh -R 127.0.0.1:47335 127.0.0.1:47335 User@192.168.0.90/`

**Reply by comfriends** [reply]
mindsdb and clickhouse are on the same server (192.168.0.90)
it is just working on each docker

**Reply by ZoranPandovski** [reply]
So I think the problem is networking and accessing the 127.0.0.1 from different containers. Can you try with the publicly accessible IP address instead of 127.0.0.1? @StpMax do you have any other ideas why this fails?

**Reply by Lee-stella** [reply]
Yes. Another container.
Clickhouse and mindsdb can communicate over a network named app. (ex) app_mindsdb & app_clickhouse
So, if I write "app_mindsdb" in the host of mysql, does it solve it?


        "mysql": {
            "database": "mindsdb",
            "host": "app_mindsdb",
            "password": "",
            "port": "47335",
            "ssl": true,
            "user": "mindsdb"
        }

**Reply by Lee-stella** [reply]
@ZoranPandovski We look forward to your reply.

**Reply by ZoranPandovski** [reply]
Sorry for the late response. I think yes, you can try with the name `app_mindsdb` or the public IP address of `app_mindsdb` container.
